src/index/silentpaymentindex.cpp

```diff
@@ -16,6 +16,7 @@
 #include <hash.h>
 
 constexpr uint8_t DB_SILENT_PAYMENT_INDEX{'s'};
+const int TAPROOT_MAINNET_ACTIVATION_HEIGHT{709632};
```

Comment:
Why hardcode this value here when we can get it from chain params?

Reply:
Because I want to drop that in #26201. Though perhaps Silent Payments itself is a reason to keep it.

Reply:
(though I think the specific reason here was because I was in a hurry)

Reply:
Ok, I see. If the taproot activation height is removed we decide not to use that specific height at all but rather one that is around the time the BIP was proposed or even to some time around now? The specifics of the approach have been changed a lot and so even if someone already has experimented with it in the last few months the current iteration likely won't work for them to be able to spend their funds, no @josibake ? Had we discussed adding the start height as a parameter, that may be another option, though raising the complexity quite a bit...